### PR TITLE
[pkg/batchperresourceattr] Add CODEOWNERS

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** <Describe what has changed.>

Add CODEOWNERS for `pkg/batchperresourceattr`. 
I chose these CODEOWNERS based on the [known importers](https://pkg.go.dev/github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchperresourceattr?tab=importedby) of the module.

**Link to tracking Issue:** Updates #3870

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

File: .github/CODEOWNERS
Comment (dmitryax) [inline .github/CODEOWNERS:113]:
```suggestion
pkg/batchperresourceattr/                            @open-telemetry/collector-contrib-approvers @atoulme @pmcollins @dmitryax
```
Owais is not actively involved in OTel now

**Review by bogdandrutu — APPROVED** [review]

**Comment by mx-psi** [comment]
@pmcollins would love to have your approval before merging this :)

**Comment by bogdandrutu** [comment]
@mx-psi @pmcollins is not that active anymore in this repo :) I know from some internal sources :))

**Comment by mx-psi** [comment]
> @mx-psi @pmcollins is not that active anymore in this repo :) I know from some internal sources :))

That makes me then the only Pablo now contributing to the Collector I think :smile:

**Comment by pmcollins** [comment]
Not sure where all of this is coming from 😆. I do have this PR out: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/15233. BTW if you're reading this, reviews are appreciated 🙂.

**Comment by mx-psi** [comment]
> Not sure where all of this is coming from laughing. I do have this PR out: #15233. BTW if you're reading this, reviews are appreciated slightly_smiling_face.

Oh okay, I haven't gotten around to checking that out yet, didn't realize it was your PR :) 

Would you want to be a CODEOWNER for this module then?

**Comment by pmcollins** [comment]
> > Not sure where all of this is coming from laughing. I do have this PR out: #15233. BTW if you're reading this, reviews are appreciated slightly_smiling_face.
> 
> Oh okay, I haven't gotten around to checking that out yet, didn't realize it was your PR :)
> 
> Would you want to be a CODEOWNER for this module then?

Oh, no, thank you -- I'm not familiar with it. Just wanted to chime in and say that I'm still around (but, as Bogdan suggests, working on other things, too :)

**Pull request merged**
